src/hotspot/share/opto/loopnode.cpp

```diff
@@ -4182,22 +4182,41 @@ bool PhaseIdealLoop::process_expensive_nodes() {
 #ifdef ASSERT
 bool PhaseIdealLoop::only_has_infinite_loops() {
```

Comment:
> This time it goes over all children of the root of the loop
tree, collects all controls for the loop and its inner loop. It then
checks whether any control is a branch out of the loop and if it is
whether it's not a NeverBranch.

Maybe you can add this summary as a comment here.